presto-main/src/main/java/com/facebook/presto/operator/aggregation/builder/InMemoryHashAggregationBuilder.java

```diff
@@ -108,30 +102,48 @@ public boolean checkFullAndUpdateMemory()
         }
     }
 
-    public Iterator<Page> buildResult()
+    public long getSizeInMemory()
```

Comment:
I will use both of those methods in spillable implementations. Both `SpillableHashAggregationBuilder` and `MergingHashAggregationBuilder` will hold a reference to `InMemoryHashAggregationBuilder` and they will use those method. And since those methods are not required by `HashAggregationOperator` there is no point making them part of the interface.
 
That's an artifact after completely restructuring this PR. At first I have written spilling PR and after that I have refactored the code. Rearranging those two things, so that refactor comes first was tedious and it left some things that could have been done differently. For example those methods could be indeed private and later could be changed to public.